android/expoview/src/main/java/versioned/host/exp/exponent/modules/api/ScreenOrientationModule.java

```diff
@@ -55,47 +60,212 @@ public void onCatalystInstanceDestroy() {
+    } catch (Exception e) {
+      promise.reject(GENERIC_ANDROID_ERROR, e.toString());
       return;
     }
-    promise.resolve(true);
+    promise.resolve(null);
+
+  }
+
+  @ReactMethod
+  public void unlockAsync(Promise promise) {
+    lockAsync(DEFAULT, promise);
+  }
+
+  @ReactMethod
+  public void getOrientationAsync(Promise promise) {
+    Activity activity = getCurrentActivity();
+    if (activity == null) {
+      promise.reject(ANDROID_NO_ACTIVITY, ANDROID_NO_ACTIVITY_MSG);
+      return;
+    }
+
+    try {
+      promise.resolve(getScreenOrientation(activity));
+    } catch (IllegalStateException e) {
+      // We don't know what the screen orientation is from surface rotation
+      promise.resolve(UNKNOWN);
+    } catch (Exception e) {
+      promise.reject(GENERIC_ANDROID_ERROR, e.toString());
+    }
+  }
+
+  @ReactMethod
+  public void getOrientationLockAsync(Promise promise) {
+    Activity activity = getCurrentActivity();
+    if (activity == null) {
+      promise.reject(ANDROID_NO_ACTIVITY, ANDROID_NO_ACTIVITY_MSG);
+      return;
+    }
+
+    int orientationAttr = activity.getRequestedOrientation();
+    promise.resolve(convertToOrientationLock(orientationAttr));
+  }
+
+  @ReactMethod
+  public void getOrientationLockPlatformAsync(Promise promise) {
+    Activity activity = getCurrentActivity();
+    if (activity == null) {
+      promise.reject(ANDROID_NO_ACTIVITY, ANDROID_NO_ACTIVITY_MSG);
+      return;
+    }
+
+    promise.resolve(activity.getRequestedOrientation());
+  }
+
+  // https://stackoverflow.com/questions/10380989/how-do-i-get-the-current-orientation-activityinfo-screen-orientation-of-an-a
+  // Will not work in all cases as surface rotation is not standardized across android devices, but this is best effort
+  private String getScreenOrientation(Activity activity) throws IllegalStateException {
+    WindowManager windowManager = activity.getWindowManager();
+    int rotation = windowManager.getDefaultDisplay().getRotation();
+    DisplayMetrics dm = new DisplayMetrics();
+    windowManager.getDefaultDisplay().getMetrics(dm);
+    int width = dm.widthPixels;
+    int height = dm.heightPixels;
+    String orientation;
+    // if the device's natural orientation is portrait:
+    if ((rotation == Surface.ROTATION_0
+        || rotation == Surface.ROTATION_180) && height > width ||
+        (rotation == Surface.ROTATION_90
+            || rotation == Surface.ROTATION_270) && width > height) {
+      switch (rotation) {
+        case Surface.ROTATION_0:
+          orientation = PORTRAIT_UP;
+          break;
+        case Surface.ROTATION_90:
+          orientation = LANDSCAPE_LEFT;
+          break;
+        case Surface.ROTATION_180:
+          orientation = PORTRAIT_DOWN;
+          break;
+        case Surface.ROTATION_270:
+          orientation = LANDSCAPE_RIGHT;
+          break;
+        default:
+          throw new IllegalStateException("Unknown screen orientation.");
```

Comment:
Seems like returning UNKNOWN would be simpler here and for the sole caller that converts the caught exception to UNKNOWN anyway.